mypy/fastparse.py

```diff
@@ -345,6 +369,14 @@ def do_func_def(self, n: Union[ast3.FunctionDef, ast3.AsyncFunctionDef],
             return_type = TypeConverter(self.errors, line=n.returns.lineno
                                         if n.returns else n.lineno).visit(n.returns)
 
+            docstring_hook = self.plugin.get_docstring_parser_hook()
+            if docstring_hook is not None and not any(arg_types) and return_type is None:
```

Comment:
I wonder if there'd be value in parsing the docstring (if present) even if there are types in the signature, and flagging discrepancies (only when both are in direct conflict) and maybe even merging the info (if some args have a type only in one of the two sources)? (Admittedly we don't do that for PEP-3107-style annotations and signature type comments either, but there we at least reject the presence of both. Here you seem to completely disregard the docstring if any types are present in the signature.)